DataReader/Controller/ViewController.swift

```diff
@@ -16,20 +16,17 @@ class ViewController: UIViewController {
     @IBOutlet weak var tableView: UITableView!
     @IBOutlet weak var activitiIndicator: UIActivityIndicatorView!
     
-    private lazy var fetchedResultsController: NSFetchedResultsController<DataText> = {
-        let fetchRequest: NSFetchRequest<DataText> = DataText.fetchRequest()
-        fetchRequest.sortDescriptors = [NSSortDescriptor(key: "content", ascending: true)]
-        let fetchedResultsController = NSFetchedResultsController(fetchRequest: fetchRequest, managedObjectContext: CoreDataStack.sharedInstance.persistentContainer.viewContext, sectionNameKeyPath: nil, cacheName: nil)
-        fetchedResultsController.delegate = self
-        
-        return fetchedResultsController
-    }()
-    
+    var apiService: APIServiceProtocol! = APIService()
+    var coreDataStack: CoreDataStackProtocol! = CoreDataStack()
     
     // MARK: - View related
     
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        assert(coreDataStack != nil, "dataProvider is not allowed to be nil at this point")
```

Comment:
I'm not a fan of asserts  ;) you should handle it more gentle instead crash

Reply:
hm.. removed assert, done?